src/runtime/HalideBuffer.h

```diff
@@ -1610,6 +1610,12 @@ class Buffer {
     }
     // @}
 
+    bool all(not_void_T val) {
```

Comment:
Conceptually this sounds good; some specific concerns:
- method should be const
- name is a little too generic. `all_equal`? Anyone else have suggestions?

Reply:
I'm not sure I can make this const as for_each_value is not const.